models/classes/taskQueue/TaskLog/Broker/AbstractTaskLogBroker.php

```diff
@@ -0,0 +1,208 @@
+    public function getStatus(string $taskId): string
+    {
+        $qb = $this->getQueryBuilder()
+            ->select(self::COLUMN_STATUS)
+            ->from($this->getTableName())
+            ->andWhere(self::COLUMN_ID . ' = :id')
+            ->setParameter('id', $taskId);
+
+        return (string)$qb->execute()->fetchColumn();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    abstract public function getTableName(): string;
+
+    /**
+     * @inheritdoc
+     */
+    abstract public function add(TaskInterface $task, string $status, string $label = null): void;
+
+    /**
+     * @inheritdoc
+     */
+    abstract public function updateStatus(string $taskId, string $newStatus, string $prevStatus = null): int;
+
+    /**
+     * @inheritdoc
+     */
+    abstract public function addReport(string $taskId, Report $report, string $newStatus = null): int;
+
+    /**
+     * @inheritdoc
+     * @throws common_exception_Error
+     */
+    public function getReport(string $taskId): ?Report
+    {
+        $qb = $this->getQueryBuilder()
+            ->select(self::COLUMN_REPORT)
+            ->from($this->getTableName())
+            ->andWhere(self::COLUMN_ID . ' = :id')
+            ->setParameter('id', $taskId);
+
+        if (
+            ($reportJson = $qb->execute()->fetchColumn())
+            && ($reportData = json_decode($reportJson, true)) !== null
+            && json_last_error() === JSON_ERROR_NONE
+        ) {
+            // if we have a valid JSON string and no JSON error, let's restore the report object
+            return Report::jsonUnserialize($reportData);
+        }
+
+        return null;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function search(TaskLogFilter $filter): iterable
+    {
+        try {
+            $qb = $this->getSearchQuery($filter);
+            $collection = TaskLogCollection::createFromArray(
+                $qb->execute()->fetchAll(),
+                $this->getPersistence()->getPlatForm()->getDateTimeFormatString()
+            );
+        } catch (Exception $exception) {
+            $this->logError('Searching for task logs failed with MSG: ' . $exception->getMessage());
+            $collection = TaskLogCollection::createEmptyCollection();
+        }
+
+        return $collection;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function count(TaskLogFilter $filter): int
+    {
+        try {
+            $qb = $this->getCountQuery($filter);
+            return (int)$qb->execute()->fetchColumn();
+        } catch (Exception $e) {
+            $this->logError('Counting task logs failed with MSG: ' . $e->getMessage());
+        }
+
+        return 0;
+    }
+
+    private function getSearchQuery(TaskLogFilter $filter): QueryBuilder
+    {
+        $qb = $this->getQueryBuilder()
+            ->select($filter->getColumns())
+            ->from($this->getTableName());
+
+        $qb->setMaxResults($filter->getLimit());
+        $qb->setFirstResult($filter->getOffset());
```

Comment:
```suggestion
        $qb = $this->getQueryBuilder()
            ->select($filter->getColumns())
            ->from($this->getTableName())
            ->setMaxResults($filter->getLimit())
            ->setFirstResult($filter->getOffset());
```
We can get leverage from the fluent interface unless we cannot use it.